docs/app/services/app-configuration/app-configuration.service.ts

```diff
@@ -3,19 +3,39 @@ import { Injectable } from '@angular/core';
 @Injectable()
 export class AppConfiguration {
 
-    private config = this.getConfig();
+    private _data = {
+        'config': require('../../data/config.json'),
+        'config.dev': require('../../data/config.dev.json'),
+        'footer-navigation': require('../../data/footer-navigation.json'),
+        'landing-page': require('../../data/landing-page.json'),
+        'team-page': require('../../data/team-page.json'),
+        'top-navigation': require('../../data/top-navigation.json'),
+        'components-page': require('../../data/components-page.json'),
+        'css-page': require('../../data/css-page.json'),
+        'charts-page': require('../../data/charts-page.json'),
+    };
+
+    public documentationPages = ['components-page', 'css-page', 'charts-page'];
 
     get(key: string): any {
         return key.split('.').reduce((prev, curr) => {
             return prev ? prev[curr] : undefined;
-        }, this.config);
+        }, this.getConfig());
+    }
+
+    getConfigurationData(key: string) {
+        return this._data[key];
+    }
+
+    setConfigurationData(key: string, data: any) {
+        this._data[key] = data;
     }
 
     private getConfig() {
         switch (process.env.ENV) {
             case 'development':
-                return require('../../data/config.dev.json');
+                return this._data['config.dev'];
         }
-        return require('../../data/config.json');
+        return this._data['config'];
```

Comment:
I'm not sure but could this use dot notation? Or is bracket notation preferred here?

Reply:
It's a key-value table, so dot is not preferred. Most keys include a period or hyphen so using property syntax would be impossible.

Reply:
OK I didn't know that, thanks.